build/scripts/build.ps1

```diff
@@ -636,6 +636,31 @@ function Stop-VSProcesses() {
     Get-Process -Name "devenv" -ErrorAction SilentlyContinue | Stop-Process
 }
 
+# Manually verify that the correct SDK is installed. This is necessary to guard 
+# against MSBuild silently downgrading to a different SDK.
+#
+# Issue tracking removing this check: https://github.com/dotnet/roslyn/issues/25378
+# Relevant issues:
+#   https://github.com/dotnet/announcements/issues/57
+#   https://github.com/dotnet/core-setup/issues/3805
+#   https://github.com/dotnet/core-setup/issues/3807
+function Check-BadDotnetSdkVersion() {
+    $all = Exec-command $dotnet "--list-sdks"
+    $sdkVersion = Get-ToolVersion "dotnetSdk"
+    $foundVersion = $false
+    foreach ($version in $all) { 
+        if ($version -match "^$sdkVersion") {
+            $foundVersion = $true
+        }
+    }
+
+    if (-not $foundVersion) { 
+        Write-Host -ForegroundColor Red "Error! Need to install the $sdkVersion version of the .NET SDK"
+        Write-Host -ForegroundColor Red "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$($sdkVersion)/dotnet-sdk-$($sdkVersion)-win-x64.exe"
```

Comment:
I'm used to powershell having strange escaping, but the `$($(` stutter here looks like a copy/paste error?

Reply:
@jasonmalinowski the actual use here is `$($` and it's necessary-ish. Powershell allows for `-` to be a part of variable names hence the `$(...)` here makes it clear that I want the interpolation to be `$sdkVersion` vs. `$sdkVersion-win-x64`. 

Most uses of `-` seem to require the `{ }` syntax as well but it's hard to track down a good referece. In either case I feel safer disambiguating.

As to why i did it for `/ /` as well ... just consistency within a line. Don't feel strongly about it.

Reply:
Oh must have misread then.